[DataGrid] Allow to customize the columns exported as CSV (#2008)
mui-org/material-ui-x: docs/src/pages/components/data-grid/export/export.md

{{"demo": "pages/components/data-grid/export/ExportSelectorGrid.js", "bg": "inline"}}

> ### Only export some columns

Reviewer: Developers can opt to export more columns than only those that are visible.

```suggestion
### Customize which columns are exported
```

> ### Customize which columns are exported

Reviewer 2: To keep the headers short?

```suggestion
### Customize exported columns
```

> By default, the CSV will only contain the visible columns of the table.

Reviewer: ```suggestion
By default, the CSV will only contain the visible columns of the grid.
```

> 

Reviewer 2: No real need for two `<p>` here:

```suggestion
```

To include or hide other columns, there are two ways:

1. Define the exact columns to be exported with the `fields` attribute in the `csvOptions` prop of [`GridToolbarExport`](/components/data-grid/components/#toolbar).

>    Set `allColumns` in `csvOptions` to true to include hidden columns, instead of only the visible ones.

Reviewer 2: I think that we could make it even more obvious for the developers:

````suggestion
    ```jsx
   <GridToolbarExport csvOptions={{ fields: ['id', 'name'] }} />
   ```

   Set `allColumns` in `csvOptions` to true to include hidden columns, instead of only the visible ones.
````

> 
> To export all columns, including the hidden one, you can use the `allColumns` option in the `csvOptions` property of `GridToolbarExport`
> 
> To exclude additional columns, you can use the `disableExport` property available in `GridColDef`
> 
> To define the exact columns you want to export, you can use the `fields` option in the `csvOptions` property of `GridToolbarExport`

Reviewer: I think is missing which approach takes precedence. We have two main ways: `fields` and `disableExport`. If `fields` is false, then developers can also use `allColumns`.

```suggestion
To include or hide other columns, there are two ways:

1. Define the exact columns to be exported with the `fields` attribute in the `csvOptions` prop of [`GridToolbarExport`](/components/data-grid/components/#toolbar). 
Set `allColumns` in `csvOptions` to true to include hidden columns, instead of only the visible ones.

2. Set the `disableExport` attribute to true in each `GridColDef`.

<!-- demo -->
```

> To define the exact columns you want to export, you can use the `columnKeys` option in the `csvOptions` property of `GridToolbarExport`

Reviewer 3: columnKeys => keys are called fields in `GridColDef`, I think `keys` is breaking the ubiquitous language.

Author: I took the naming from AG Grid but field would indeed match better our own naming.

Author: Perhaps we should also rename `allColumns` into `allFields`

> 2. Set the `disableExport` attribute to true in each `GridColDef`.

Reviewer 2: ````suggestion
2. Set the `disableExport` attribute to true in each `GridColDef`.
  
  ```jsx
  <DataGrid
    columns={[
      { field: 'id', disableExport: true },
      { field: 'brand' },
    ]}
  ```
````
